### ISSUE TITLE
`properties` property in behavior overrides getter on prototype

### ISSUE THREAD

**Issue by rictic** [Polymer/polymer#3410]
A property defined in the `properties` block in a behavior will overwrite a getter on an element prototype. This is unexpected, as we expect the prototype to win.

jsbin: https://jsbin.com/hevimo/edit?html,output

The key bit:

``` html
<dom-module id="example-elem">
  <template>
    {{x}}
  </template>
</dom-module>
<script>
  var Behave = {
    properties: {
      x: {
        value: 'behavior property',
        notify: true
      }
    }
  }

  Polymer({
    is: 'example-elem',

    behaviors: [Behave],

    get x() {
      console.log('never called');
      return 'prototype getter';
    }
  });
</script>
```

**Comment by javiervelezreyes** [comment]
Right Pit, as a possible workaround It could be a good idea to include get & set functions within the property definition. Thus Polymer could  under the hood adds its get/set logic as a decoration to be executed before and/or after the invocations to get/set functions defined in the properties object. From the developer point of view:

    properties: {
      x: {
           value: 'behavior property',
           notify: true,
           get: function () { .... },
           set: function (v) { .... }
      }
    }
 
The auto-generated get/set by Polymer:

      get x () {
           // notifify, reflectToAtribute & other polymer logic
           return properties.x.get.call (...);
      }
      set x (v) {
           // notifify, reflectToAtribute & other polymer logic
           return properties.x.set (... v);
      }

**Comment by kevinpschaaf** [comment]
Polymer's assumption is that `properties` with side effects are fully managed by Polymer's metaprogramming system, and thus cannot also be managed via user accessors.

We've experimented with attempting to add support for deferring to a user's accessor various times.  This is ultimately problematic for two reasons:
1. Avoiding stomping on a user accessor requires an expensive prototype walk calling `getOwnPropertyDescriptor` and checking for a user getter/setter for every managed property
2. Polymer's data-binding system detects when binding from one Polymer element to another, and bypasses the accessor altogether (and instead calls Polymer internal methods for enqueuing property changes) for batching properties. In this case, the user's desired behavior of the accessor being called wouldn't occur.

As such, with the current design in `Polymer.Element` we don't think we can support this without breaking compatibility.

However in Polymer's lowest-level `PropertiesChanged` mixin that will be released in v2.4, accessors are not overridden if they already exist and can instead call `_setProperty` and `_getProperty` API's within user accessors to implement user code in accessors, so future base classes built on the lower level `PropertiesChanged` can support this pattern.  Just not in `Polymer.Element`, due to compatibility concerns.
